package-name-disputes.md

```diff
@@ -0,0 +1,131 @@
+
+## Description
+
+In an ecosystem of millions of daily users and more than 250,000
+packages, one user may wish to use a package name that another user
+has already published under.  Here are some common ways that happens
+(each of these is based on actual events.)
+
+1. Alice writes a JavaScript module `foo`, which is not node-specific.
+   Alice doesn't use node at all.  Yusuf wants to use `foo` in node, so he
+   wraps it in an npm module.  Some time later, Alice starts using node,
+   and wants to take over management of her program.
+2. Yusuf writes an npm module `foo`, and publishes it.  Perhaps much
+   later, Alice finds a bug in `foo`, and fixes it.  She sends a pull
+   request to Yusuf, but Yusuf doesn't have the time to deal with it,
+   because he has a new job and a new baby and is focused on his new
+   Erlang project, and kind of not involved with node any more.  Alice
+   would like to publish a new `foo`, but can't, because the name is
+   taken.
+3. Yusuf writes a 10-line flow-control library, and calls it `foo`, and
+   publishes it to the npm registry.  Being a simple little thing, it
+   never really has to be updated.  Alice works for Foo Inc, the makers
+   of the critically acclaimed and widely-marketed `foo` JavaScript
+   toolkit framework.  They publish it to npm as `foojs`, but people
+   are routinely confused when `npm install foo` is some different
+   thing.
+4. Yusuf writes a parser for the widely-known `foo` file format, because
+   he needs it for work.  Then, he gets a new job, and never updates
+   the prototype.  Later on, Alice writes a much more complete `foo`
+   parser, but can't publish, because Yusuf's `foo` is in the way.
+
+The validity of Alice's claim in each situation can be debated.
+However, Alice's appropriate course of action in each case is the same.
+
+1. `npm owner ls foo`.  This will tell Alice the email address of the
+   owner (Yusuf).
+2. Alice emails Yusuf:
+    - explaining the terms of her claim to the package name
+    - what she would like to do with the module name
+    - CC's npm support staff <support@npmjs.com>
+    - if appropriate, suggests that Yusuf can run `npm owner add
+      alice foo` to add Alice as an owner of the `foo` package.
+3. After a reasonable amount of time, if Yusuf has not responded, or if
+   Yusuf and Alice can't come to any sort of resolution, support
+   <support@npmjs.com> will follow up to help moderate and move the
+   conversation. ("Reasonable" is usually at least 4 weeks.)
+
+## Assumptions and Values 
+
+These are the base assumptions and values by which naming disputes will
+be evaluated by the npm team:
+
+- A package with a given name should be the package that the most
+  people expect it to be, this means:
+
+    - All things being equal, a package should remain the same package
+    - A popular package name will always point to the same package
+    - A popular trademark will point to an associated package
```

Comment:
A trademark should belong to the owner of that trademark, but we require the owner of the trademark to provide proof that they own it.